packages/protocol/test/ACE/validators/publicRange/index.js

```diff
@@ -0,0 +1,393 @@
+    return { originalNote, utilityNote, publicComparison, isGreaterOrEqual };
+};
+
+const getDefaultLessThanNotes = async () => {
+    const isGreaterOrEqual = false;
+    const originalNoteValue = 10;
+    const utilityNoteValue = 30;
+    const publicComparison = 20;
+    const { originalNote, utilityNote } = await getNotes(originalNoteValue, utilityNoteValue);
+    return { originalNote, utilityNote, publicComparison, isGreaterOrEqual };
+};
+
+contract('Public range validator', (accounts) => {
+    const sender = accounts[0];
+
+    before(async () => {
+        publicRangeValidator = await PublicRange.new({ from: sender });
+    });
+
+    describe('greater than tests', () => {
+        describe('Success states', () => {
+            it('should validate public range proof when given explicit utilityNote', async () => {
+                const { originalNote, utilityNote, publicComparison, isGreaterOrEqual } = await getDefaultGreaterThanNotes();
+                const proof = new PublicRangeProof(originalNote, publicComparison, sender, isGreaterOrEqual, utilityNote);
+                const data = proof.encodeABI();
+                const result = await publicRangeValidator.validatePublicRange(data, sender, bn128.CRS, { from: sender });
+                expect(result).to.equal(proof.eth.outputs);
+            });
+
+            it('should validate public range proof for note of zero value', async () => {
+                const isGreaterOrEqual = true;
+                const originalNoteValue = 10;
+                const utilityNoteValue = 0;
+                const publicComparison = 10;
+                const { originalNote, utilityNote } = await getNotes(originalNoteValue, utilityNoteValue);
+                const proof = new PublicRangeProof(originalNote, publicComparison, sender, isGreaterOrEqual, utilityNote);
+                const data = proof.encodeABI();
+                const result = await publicRangeValidator.validatePublicRange(data, sender, bn128.CRS);
+                expect(result).to.equal(proof.eth.outputs);
+            });
+
+            it('should validate public range proof with challenge that has group modulus added to it', async () => {
+                const { originalNote, utilityNote, publicComparison, isGreaterOrEqual } = await getDefaultGreaterThanNotes();
+                const proof = new PublicRangeProof(originalNote, publicComparison, sender, isGreaterOrEqual, utilityNote);
+                proof.challenge = proof.challenge.add(bn128.groupModulus);
+                proof.constructOutputs(); // why do we have to do this?
+                const data = proof.encodeABI();
+                const result = await publicRangeValidator.validatePublicRange(data, sender, bn128.CRS);
+                expect(result).to.equal(proof.eth.outputs);
+            });
+        });
+
+        describe('Failure states', () => {
+            it('should fail if notes do NOT balance', async () => {
```

Comment:
We should check the failure states when the input or utility note is created over an incorrect trusted setup private key.

Reply:
Yep agreed. This needs doing for all proofs, assigned a seperate task to do this